avaframe/in4Region/splitInputs.py

```diff
@@ -0,0 +1,876 @@
+            continue
+            
+        # Clip the DEM data
+        clippedData = raster[rowStart:rowEnd, colStart:colEnd]
+        
+        # Create header for clipped DEM
+        clippedHeader = header.copy()
+        clippedHeader['ncols'] = colEnd - colStart
+        clippedHeader['nrows'] = rowEnd - rowStart
+        clippedHeader['xllcenter'] = xOrigin + colStart * cellSize
+        clippedHeader['yllcenter'] = yOrigin + rowStart * cellSize
+        
+        # Update transformation matrix for clipped DEM
+        newTransform = rasterio.transform.from_origin(
+            clippedHeader['xllcenter'] - (cellSize/2),  # Convert center to corner
+            clippedHeader['yllcenter'] + (clippedHeader['nrows'] * cellSize) - (cellSize/2),  # Convert to top-left y
+            cellSize,
+            cellSize
+        )
+        clippedHeader['transform'] = newTransform
+        
+        # Write clipped DEM
+        outputDEM = outputDir / dirName / 'Inputs' / f"{dirName}_DEM"
+        rasterUtils.writeResultToRaster(clippedHeader, clippedData, outputDEM, flip=True)
+        log.debug(f"Clipped DEM saved to: {outputDEM}")
+
+        # Store DEM extents (reduced by one pixel on each side to ensure DEM > clip extents)
+        xMinDEM = clippedHeader['xllcenter'] - (cellSize/2) + cellSize
+        yMinDEM = clippedHeader['yllcenter'] - (cellSize/2) + cellSize
+        xMaxDEM = clippedHeader['xllcenter'] + (clippedHeader['ncols'] * cellSize) - (cellSize/2) - cellSize
+        yMaxDEM = clippedHeader['yllcenter'] + (clippedHeader['nrows'] * cellSize) - (cellSize/2) - cellSize
+        groupExtents[dirName] = (xMinDEM, xMaxDEM, yMinDEM, yMaxDEM)
+    
+    return groupExtents
+
+def clipAndMoveOptionalInput(inputDir, outputDir, groupExtents):
+    """Clip and move ENT and RES files based on group DEM extent.
+    
+    #ToDo: extend to include other input types
+    
+    Parameters
+    ----------
+    inputDir : pathlib.Path
+        Path to input directory containing ENT and RES folders
+    outputDir : pathlib.Path
+        Path to output directory where clipped files will be saved
+    groupExtents : dict
+        Dictionary with dirName as key and (xMin, xMax, yMin, yMax) as value,
+        containing the DEM clipping extents for each group
+
+    Returns
+    -------
+    groupFeatures : dict
+        Dictionary containing clipped features for each group and type
+        {dirName: {'ENT': [...], 'RES': [...]}}
+    """
+    groupFeatures = {}
+    # Process ENT and RES directories
+    for dirType in ['ENT', 'RES']:
+        typeDir = inputDir / dirType
+        if not typeDir.exists():
+            log.warning(f"No {dirType} directory found in {inputDir}")
+            continue
+
+        # Find shapefile in directory
+        shpFile = next(typeDir.glob("*.shp"), None)
+        if not shpFile:
+            log.warning(f"No shapefile found in {typeDir}")
+            continue
```

Comment:
again this could already be available if you use getInputCom1DFA once to create inputSimFiles dict